rclpy/rclpy/action/server.py

```diff
@@ -362,7 +362,15 @@ async def _execute_cancel_request(self, request_header_and_message):
 
             if CancelResponse.ACCEPT == response:
                 # Notify goal handle
-                goal_handle._update_state(_rclpy.GoalEvent.CANCEL_GOAL)
+                try:
+                    # If the goal's just scceeded after user cancel callback
+                    # that will generate exception from invalid transition.
+                    goal_handle._update_state(GoalEvent.CANCEL_GOAL)
+                except Exception as ex:
+                    self._node.get_logger().error(
+                        'Failed to cancel goal in cancel callback: {0}'.format(ex))
```

Comment:
do we have to log an error here?
This can be a debug message really, receiving a cancel goal after the goal succeeded is normal behavior.

Reply:
good point, it can be debug message.